homeassistant/components/switcher_kis/light.py

```diff
@@ -39,12 +39,29 @@ def async_add_light(coordinator: SwitcherDataUpdateCoordinator) -> None:
             coordinator.data.device_type.category
             == DeviceCategory.SINGLE_SHUTTER_DUAL_LIGHT
         ):
-            entities.extend(SwitcherDualLightEntity(coordinator, i) for i in range(2))
+            entities.extend(SwitcherMultiLightEntity(coordinator, i) for i in range(2))
         if (
             coordinator.data.device_type.category
             == DeviceCategory.DUAL_SHUTTER_SINGLE_LIGHT
         ):
             entities.append(SwitcherSingleLightEntity(coordinator, 0))
+        if coordinator.data.device_type.category == DeviceCategory.LIGHT:
+            if coordinator.data.device_type == DeviceType.LIGHT_SL03:
+                entities.extend(
+                    SwitcherMultiLightEntity(coordinator, i) for i in range(3)
+                )
+            if coordinator.data.device_type in (
+                DeviceType.LIGHT_SL02,
+                DeviceType.LIGHT_SL02_MINI,
+            ):
+                entities.extend(
+                    SwitcherMultiLightEntity(coordinator, i) for i in range(2)
+                )
+            if coordinator.data.device_type in (
+                DeviceType.LIGHT_SL01,
+                DeviceType.LIGHT_SL01_MINI,
+            ):
+                entities.append(SwitcherSingleLightEntity(coordinator, 0))
```

Comment:
This starts to be too complex. You can create a constant mapping (`dict`) that sets the number of lights for each type so we will have a simple code here that adds the correct number of entities by device type.

Reply:
It is done..
Should I move also the Runenr S11 and S12 to the dict and make it all one block?

Reply:
Also, should I do the same for the cover.py?

Reply:
> It is done.. Should I move also the Runenr S11 and S12 to the dict and make it all one block?

Yes

> Also, should I do the same for the cover.py?

Better to do it on a different PR

Reply:
It is done.
next PR we will do for cover.py